pkg/monitor/cluster/nodes.go

```diff
@@ -0,0 +1,57 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type metricsLabel string
+
+const (
+	notReadyNodes           metricsLabel = "notReadyNodes"
+	memoryPressureNodes     metricsLabel = "memoryPressureNodes"
+	diskPressureNodes       metricsLabel = "diskPressureNodes"
+	pidPressureNodes        metricsLabel = "pidPressureNodes"
+	networkUnavailableNodes metricsLabel = "networkUnavailableNodes"
+)
+
+var metricsLabelMap = map[corev1.NodeConditionType]metricsLabel{
+	corev1.NodeReady:              notReadyNodes,
+	corev1.NodeMemoryPressure:     memoryPressureNodes,
+	corev1.NodeDiskPressure:       diskPressureNodes,
+	corev1.NodePIDPressure:        pidPressureNodes,
+	corev1.NodeNetworkUnavailable: networkUnavailableNodes,
+}
+
+func (mon *Monitor) emitNodesMetrics() error {
+	nodes, err := mon.cli.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	mon.emitGauge("cluster.nodes.total", int64(len(nodes.Items)), map[string]string{})
+
+	counters := map[metricsLabel]int64{}
+	for _, node := range nodes.Items {
+		for _, c := range node.Status.Conditions {
+			// count 'Unknown' status as unhealthy state for each condition. In this way
+			// we can flag issues without creating additional timeseries for each condition.
+			// for NodeReady count a node when the status is False (not ready) or Unknown
+			// for other conditions count when the status is True or Unknown
+			if c.Type == corev1.NodeReady && c.Status != corev1.ConditionTrue ||
+				c.Type != corev1.NodeReady && c.Status != corev1.ConditionFalse {
+				counters[metricsLabelMap[c.Type]] += 1
```

Comment:
do `counters[string(c.Type)]++` except for the case of NodeReady, in which case do `counters["NotReady"]++` ?

Reply:
this is another way of doing this, but can you tell me why is this better over what I suggest? At least how I see it - I describe the condition in one place in two "sentences" then I increment all the counters in the same manner.

Reply:
I think the answer might be half way between us.  I was worried about the case of c.Type not being in metricsLabelMap, which actually your PR is handling correctly currently, but in fact I realise that a whitelist set of conditions to forward is in fact necessary.

Reply:
I see what you mean now. I'll get the update